### ISSUE TITLE
Autoflag bug in Human-Human games

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#814]
Original [issue 814](https://code.google.com/p/pychess/issues/detail?id=814) reported by [gbtami](https://code.google.com/u/gbtami/) 2013-09-08

If autoflag is on in a Human-Human game and one side is reaching 0 time, the following message appears:

The flag call returns an error
"Your opponent is not out of time."

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=814#c1) posted by [jcheetham19](https://code.google.com/u/jcheetham19/) 2014-08-10

This issue is caused by this code from offerRecieved in GameModel.py:

```
    elif offer.type == FLAG\_CALL:
        assert self.timed
        if self.timemodel.getPlayerTime(player.color) <= 0:
            if self.timemodel.getPlayerTime(1-player.color) <= 0:
                self.end(DRAW, DRAW\_CALLFLAG)
            elif not playerHasMatingMaterial(self.boards[-1], 1-player.color):
                if player.color == WHITE:
                    self.end(DRAW, DRAW\_BLACKINSUFFICIENTANDWHITETIME)
                else:
                    self.end(DRAW, DRAW\_WHITEINSUFFICIENTANDBLACKTIME)
            else:
                if player == self.players[BLACK]:
                    self.end(WHITEWON, WON\_CALLFLAG)
                else:
                    self.end(BLACKWON, WON\_CALLFLAG)
        else:
            player.offerError(offer, ACTION\_ERROR\_NOT\_OUT\_OF\_TIME)
```

I changed this and tested it.
I attached the patch and a list of the tests I did.
- attached [tests.txt](https://code.google.com/p/pychess/issues/detail?id=814#c1)
- attached [flagcall.patch](https://code.google.com/p/pychess/issues/detail?id=814#c1)

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=814#c2) posted by [gbtami](https://code.google.com/u/gbtami/) 2014-08-10

Applied, thx!

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=814#c3) posted by [gbtami](https://code.google.com/u/gbtami/) 2014-08-10

This issue was closed by revision 3353baa05b29.
- changed status to: Fixed
